MAINTAINING.md

```diff
@@ -0,0 +1,240 @@
+automatically redirect the visitor to the docs for the latest published version
+of the gem. This version is hardcoded in the HTML page, but is also updated
+automatically by the `docs:publish` and `docs:publish_latest` tasks.
+
+*\* thoughtbot owns <https://shoulda.io>, and
+they've got `matchers.shoulda.io` set up on the DNS level as an alias for
+`thoughtbot.github.io/shoulda-matchers`.*
+
+### Re-publishing docs
+
+In general you'll use the `release` task to update the docsite, but there may be
+a situation where you'll have to do it manually.
+
+You can re-publish the docs for the latest version (as governed by
+`lib/shoulda/matchers/version.rb`) by running:
+
+```bash
+bundle exec rake docs:publish_latest
+```
+
+If you want to publish the docs for a version privately but manually set what
+the docsite considers as the "latest version" (specifically around the
+auto-redirect as mentioned above), you can run this instead:
+
+```bash
+bundle exec rake docs:publish[version, latest_version]
+```
+
+Here, `version` and `latest_version` are both version strings. For instance, you
+might say:
+
+```bash
+bundle exec rake docs:publish[4.0.0, 3.7.2]
+```
+
+which would publish the docs for 4.0.0, but then set the redirect for
+<https://matchers.shoulda.io/docs> to 3.7.2.
+
+## Versioning
+
+### Naming a new version
+
+As designated in the README, we follow [SemVer 2.0][semver]. This offers a
+meaningful baseline for deciding how to name versions. Generally speaking:
+
+[semver]: https://semver.org/spec/v2.0.0.html
+
+* We bump the "major" part of the version if we're introducing
+  backward-incompatible changes (e.g. changing the API or core behavior,
+  removing parts of the API, or dropping support for a version of Ruby).
+* We bump the "minor" part if we're adding a new feature (e.g. adding a new
+  matcher or adding a new qualifier to a matcher).
+* We bump the "patch" part if we're merely including bugfixes.
+
+### Releasing a new version
+
+Releasing a new version is very simple:
+
+1. First, you'll want to be given ownership permissions for the Ruby gem itself.
+   If you want to give someone else these rights, you can use:
+
+   ```bash
+   gem owner shoulda-matchers -a <email address>
+   ```
+1. Next, you'll want to update the `VERSION` constant in
+   `lib/shoulda/matchers/version.rb`. This constant is referenced in the gemspec
+   and is used in the Rake tasks to publish the gem on RubyGems as well as
+   generate documentation.
+1. Finally, you'll want to run:
+
+   ```bash
+   rake release
+   ```
+
+   This will not only update the docsite but also push the gem to RubyGems.
+
+## Addendum: Labels
+
+In order to corral the issue and PR backlog, we've found
+[labels] to be useful for cataloguing and tracking progress purposes. Over time
+we've added quite a collection of labels. Here's a quick list:
+
+[labels]: https://github.com/thoughtbot/shoulda-matchers/labels
+
+### Labels for issues
```

Comment:
I added a section for labels as a sort of brain dump. These labels aren't set in stone, of course, and I realize it's entirely possible we have too many. I'd be curious to know what your thoughts on them are.